### ISSUE TITLE
[3.9] Field status filter does not remove unpublished

### ISSUE THREAD

**Issue by Flowman** [joomla/joomla-cms#27889]
### Steps to reproduce the issue

Create a status field, enable filtering and remove unpublished from list.

```
		<field
			name="published"
			type="status"
			label="JOPTION_SELECT_PUBLISHED"
			filter="1,2,-2,*"
			description="JOPTION_SELECT_PUBLISHED_DESC"
			onchange="this.form.submit();"
			>
			<option value="">JOPTION_SELECT_PUBLISHED</option>
		</field>
```
### Expected result

Unpublished should be removed.

### Actual result

Unpublished is still present.

### System information (as much as possible)

Joomla 3.9.15
PHP 7.3.2

### Additional comments

Issue is in JFormFieldPredefinedList where the in_array will always return true on 0.

**Comment by brianteeman** [comment]
Where are you using this?

**Comment by Flowman** [comment]
In my own component. But I like to use Joomla native code.

It' a list of affiliates but they should only be able to be published, archived, trashed.

Either way there is a filter function on the status field or JFormFieldPredefinedList that is not working with 0.

**Comment by brianteeman** [comment]
It must be an issue in your code because clearly we use that everywhere in Joomla

**Comment by SharkyKZ** [comment]
I can confirm this issue.

**Comment by brianteeman** [comment]
so how does it work in core?

**Comment by richard67** [comment]
As far as I know (but I might be wrong, so correct me if so), the "filter" is applied at form validation, i.e. it has no impact on which options are shown but only on which options are being accepted on validation.

**Comment by SharkyKZ** [comment]
@brianteeman We don't use this field with the values required to run into the issue.  But it's totally valid issue. The closest we have in core is this:

>https://github.com/joomla/joomla-cms/blob/fe798dd4bd2023047707053ae7a91a3ad5e9fa35/administrator/components/com_languages/models/forms/filter_languages.xml#L15

If you remove `0` from the list and view the dropdown, `Unpublished` option will still be listed.

@richard67 Right, we have conflicting attributes here. That's another issue. Needs to be solved before 4.0.

**Comment by alikon** [comment]
please test #28063
